packages/material-ui/src/ButtonBase/TouchRipple.js

```diff
@@ -98,173 +98,193 @@ export const styles = theme => ({
   },
 });
 
-class TouchRipple extends React.PureComponent {
-  // Used to filter out mouse emulated events on mobile.
-  // ignoringMouseDown = false;
+const TouchRipple = React.forwardRef(function TouchRipple(props, ref) {
```

Comment:
You need a `@ignore - something about internal or so` jsdoc comment here. SwitchBase should have it too.

Reply:
Yeah, I was thinking if that but I wasn’t sure if TouchRipple was an internal component

Reply:
Right we do document it. I think we can override the component resolver for react-docgen. Will take a look at it.

Reply:
Filed PR to react-docgen. We can workaround it for now.